dhcpv4/ztpv4/ztp.go

```diff
@@ -15,14 +17,36 @@ type VendorData struct {
 
 var errVendorOptionMalformed = errors.New("malformed vendor option")
 
-// ParseVendorData will try to parse dhcp4 options looking for more
-// specific vendor data (like model, serial number, etc).
-func ParseVendorData(packet *dhcpv4.DHCPv4) (*VendorData, error) {
+func parseVIVC(packet *dhcpv4.DHCPv4) (*VendorData, error) {
```

Comment:
the `parseVIVC` and `parseVendorClass` functions are not consistent, one returns an error if the option is not found and the other does not.

Maybe non of them should return an error in such case and only return an error if there is a parsing issue?

Reply:
Modified parseClassIdentifier to not return an error when vc is empty.